### ISSUE TITLE
Hotend Fan connection

### ISSUE THREAD

**Issue by jgwinner** [bigtreetech/Tango-3D-Printer-Motherboard#10]
Hi

Great board, I'm having fun with it.

I can't see where to connect the Hotend heatsink fan. NOT the part cooling fan, that's the PWM outputs. As near as I can tell it's not a PWM fan, just an on/off fan, but I don't really want to just repurpose the PWM fans or I'll run out of pins.

So, where should the heatsink cooling fan be connected to?

        == John ==

**Comment by jgwinner** [comment]
Anyone?

**Comment by Aurawin** [comment]
Hot-end heat sync is PWM FAN0.  That is what we all use for Marlin.  Most set fan to 100% but still marlin turns fan on at setpoint.  What you propose is keeping the fan on at boot.  You can do that by selecting one of the 2 12V two pin headers.  On the corner.  One white.  One yellow.  Either will power your fan.  

You can borrow a connector from some other 2pin fan and solder.  You will have a printer that has it's heatsink fan running all the time.  This will delay preheats.

**Comment by jgwinner** [comment]
Thanks!

I don't want the fan on all the time, I just want to know where to connect it, but I don't think the Heatsink fan needs PWM, just a controllable on/off.

Using Fan0 confuses me if I have more than 1 extruder.Do you put all part cooling fans on Fan1 and all heatsink fans on Fan0?

To be clear, if I used all 3 extruders, I'd need 6 fans, and we're out of PWM outputs. 3 heatsinks, and 3 part cooling fans, in addition to the 3 extruder heaters.

One way or another, we need 9 PWM outputs for a full 3 extruder setup. I actually usually print PETG in which case I rarely use a part cooling fan, but as you mention, if the heatsink fan is always on, it would delay preheats, and I do occasionally print PLA so I need a cooling fan.

First Alternative:

If I only have 2 extruders, I could use Extruder 0 and 1 for the heaters, Extruder 2 for part cooling fan 0, Fan0 for part cooling fan 1, and Fan 1 for both heatsink fans. I'd have to check Marlin to see how to do that, I'd be worried that if I had only one extruder running it might turn on the heatsink fan for extruder 1, then turn the same pin off for extruder 1.  

Second alternative: Put ALL cooing fans on Fan 0, and ALL Heatsink fans on Fan 1. Disadvantage; no multi-material. 

Third Alternative: Run J11 pins 5 and 6 to an external MOSFET(s) for Extruder 0 and 1, and use the onboard MOSFET pins for most of the fans. Rearrange Marlin of course. Can I use J11 pins 5 and 6? The PCB diagram doesn't show them connected to anything else.

Thoughts?

**Comment by jgwinner** [comment]
I'm not seeing any docs on what pin assignments should be for these. I opened the PCB with Altium, and the pins on the chip seem to be pretty different than what I expected - for example, Y-STEP seems to be 97, but in PINS_RUMBA it's 54

This is really confusing. I need to use some of the pins on EXP3 but it's not clear what PINS.H should have to use this pins.

I've checked all the docs. Please advise?

        == John ==

**Comment by jgwinner** [comment]
Anyone? I just need to know the correct pins, and what pins on the motherboard correspond to what pins in Marlin.

        == John ==

**Comment by Aurawin** [comment]
Tango is a rumba variant. Almost all of the pins are specified in the pins_rumba.h. And perhaps even inherited header file.

**Comment by jgwinner** [comment]
That's incorrect. I've built a spreadsheet. There are many pins that are not specified, and from the diagram, as I noted above, some are inconsistent. 

On the PCB Diagram, Y-STEP is pin 97, but in PINS_RUMBA it's 54.

**Comment by jgwinner** [comment]
So, what is Y-Step? Pin 97, or pin 54? What about the rest of them?

== John ==

**Comment by Aurawin** [comment]
I have no problems with the existing stepper pin mappings.  I run a Tronxy X5S with Dual Z Steppers.
I am curious to know what if any symptoms exist with respect to the Y stepper driver.

**Comment by McUtty** [comment]
Here I find how to config 2 Fan with the TANGO Board:
https://www.digitales-minden.de/wiki/bigtreetech-biqu-board-tango-fan1-enable-configuartion-h
